### ISSUE TITLE
Navigation in the in-built file picker

### ISSUE THREAD

**Issue by Valdnet** [SmartPack/PackageManager#44]
Steps:

1. Create a subdirectory in the directory.
2. Open the subdirectory with in-built file picker.
3. Go back to the directory by navigating the in-built file picker arrow at the top.

The window immediately returns to the main application window.
A directory window with a subdirectory should appear.

Apps:
Android 10
Package Manager v5.1

**Comment by sunilpaulmathew** [comment]
@Valdnet 
In fact, that was the intended behavior while coding the file picker. That is, 
1. The normal "Back" action (navigation key or whatever) will make the app to go to last screen.
2. The Back arrow closes the file picker from anywhere.

Is that confusing?

**Comment by Valdnet** [comment]
I think in this case, if there is no side menu for the application, it may stay so that the user can quickly return to the main application window.
In general, the applications have a side menu, and the arrow returns to the previous window that was visible.

Have you thought about adding a navigation side menu?

**Comment by sunilpaulmathew** [comment]
Nope, no plans to include side menu with file picker. I want to make it so simple. An alternate option is to repurpose the action of back menu with the most expected one, and added a cancel button on the bottom (or a close button on the top right side).

**Comment by Valdnet** [comment]
> Nope, no plans to include side menu with file picker.

I was thinking of a side menu for the entire app, not just for file picker. In such a menu, you can put info about the application, settings, etc.

**Comment by sunilpaulmathew** [comment]
> > Nope, no plans to include side menu with file picker.
> 
> I was thinking of a side menu for the entire app, not just for file picker. In such a menu, you can put info about the application, settings, etc.

Ahh, you mean a navigation drawer. Which is my least preference. I'm a big fan of apps with bottom navigation or tab layouts (e.g, the package details view in this app).

**Comment by Valdnet** [comment]
After making this change in commit https://github.com/SmartPack/PackageManager/commit/7784f1f0ca9f41c1d5cad63e1a62c131fb64c77c got even more interesting 😁.
For users who rarely use the system arrow at the bottom of their smartphone, they may be a bit surprised that they can't go back or enter the main directory.
I suggest adding a directory path at the top of the application so that you can select a directory from the given path.

Something like that:

![obraz](https://user-images.githubusercontent.com/47037905/132850028-1aa2d580-935d-426d-9877-1b2ec8bdec2a.png)

**Comment by sunilpaulmathew** [comment]
> After making this change in commit [7784f1f](https://github.com/SmartPack/PackageManager/commit/7784f1f0ca9f41c1d5cad63e1a62c131fb64c77c) got even more interesting 😁.
> For users who rarely use the system arrow at the bottom of their smartphone, they may be a bit surprised that they can't go back or enter the main directory.
> I suggest adding a directory path at the top of the application so that you can select a directory from the given path.
> 
> Something like that:
> 
> ![obraz](https://user-images.githubusercontent.com/47037905/132850028-1aa2d580-935d-426d-9877-1b2ec8bdec2a.png)

Okay, this is a good idea, but need some work. But, I'll look for it. Thanks again.

**Comment by sunilpaulmathew** [comment]
@Valdnet 
Not exactly like the way you suggested, I added a new option as seen in the screenshot.

![Screenshot_1643460980](https://user-images.githubusercontent.com/15515387/151662249-7ab69b4e-5cee-4d12-8e2e-3bdaff7f591e.png)

**Comment by Valdnet** [comment]
I decided to reopen the issue.

I noticed that by clicking on the "three dots" for the root directory of the internal memory, the application goes to the main window, for example: "All".
In my opinion, when the directory is the root directory, the "three dots" should be grayed out.
